src/components/Navigator/QuickNavigationModal.vue

```diff
@@ -272,6 +343,78 @@ export default {
     startingPointHook() {
       this.focusedIndex = this.totalItemsToNavigate - 1;
     },
+    async fetchSelectedSymbolData() {
+      // start a timer: if a full second has elapsed and the network request for
+      // data hasn't completed yet, indicate that the data is still loading in
+      // the actual UI — without the delay, the constant flashing of this
+      // message would be too distracting
+      this.loadingTimeout = setTimeout(() => {
+        if (this.previewState === PreviewState.loading) {
+          this.previewIsLoadingSlowly = true;
+        }
+      }, SLOW_LOADING_DELAY);
+
+      // exit early if the previwe data is already cached for this symbol as
+      // there is no additional work needed
+      if (!this.selectedSymbol || this.$cachedSymbolResults.has(this.selectedSymbol.uid)) {
+        clearTimeout(this.loadingTimeout);
+        this.previewIsLoadingSlowly = false;
+        return;
+      }
+
+      const fetchSymbolData = async (symbol) => {
```

Comment:
Do you think this can be extracted to its own method?

Reply:
It could be, although it's not used anywhere else yet, so I just kept it as a nested function for now.